docs/guides/basic-rbac.mdx

```diff
@@ -3,267 +3,307 @@ title: Implement basic Role Based Access Control (RBAC) with metadata
+    if (!checkRole('admin')) {
+      redirect('/')
+    }
+
+    return <p>This is the protected admin dashboard restricted to users with the `admin` role.</p>
   }
+  ```
 
-  return (
-    <>
-      <h1>This is the admin dashboard</h1>
-      <p>This page is restricted to users with the 'admin' role.</p>
-    </>
-  )
-}
-```
+  ### Add admin tools to find users and manage roles
 
-> [!NOTE]
-> You can modify the behavior of the helper function to meet your needs. Maybe it will return the roles that the user has, or you could create a `protectByRole()` and have that function handle the redirect.
+  You can use the `checkRole()` function along with server actions to build basic tools for finding users and managing roles.
 
-## Add admin tools to find users and add roles
+  Create a server action for managing a user's role.
 
-You can leverage the `checkRole()` function you added along with server actions to build basic tools for finding users and managing roles.
+  1. In your `app/admin/` directory, create an `_actions.ts` file.
+  1. Create a server action that sets a user's role. Use the `checkRole()` function to verify that the current user has the `admin` role. If they do, proceed to update the specified user's role using Clerk's [JavaScript Backend SDK](/docs/references/backend/user/update-user). This ensures that only administrators can modify user roles.
+  1. Create a server action that removes a user's role.
 
-Start with the server action. The `setRole()` action below will check the role of the current user using the `checkRoles()` helper to confirm that the user is an `admin`. It will then set the role for the selected user to the specified role.
+  ```ts {{ filename: 'app/admin/_actions.ts' }}
+  'use server'
 
-```ts {{ filename: 'app/admin/dashboard/_actions.ts' }}
-'use server'
+  import { checkRole } from '@/utils/roles'
+  import { clerkClient } from '@clerk/nextjs/server'
 
-import { checkRole } from '@/utils/roles'
-import { clerkClient } from '@clerk/nextjs/server'
+  export async function setRole(formData: FormData) {
+    // Check that the user trying to set the role is an admin
+    if (!checkRole('admin')) {
+      return { message: 'Not Authorized' }
+    }
 
-export async function setRole(formData: FormData) {
-  // Check that the user trying to set the role is an admin
-  if (!checkRole('admin')) {
-    return { message: 'Not Authorized' }
+    try {
+      const res = await clerkClient().users.updateUser(formData.get('id') as string, {
+        publicMetadata: { role: formData.get('role') },
+      })
+      return { message: res.publicMetadata }
+    } catch (err) {
+      return { message: err }
+    }
   }
 
-  try {
-    const res = await clerkClient().users.updateUser(formData.get('id') as string, {
-      publicMetadata: { role: formData.get('role') },
-    })
-    return { message: res.publicMetadata }
-  } catch (err) {
-    return { message: err }
+  export async function removeRoles(formData: FormData) {
```

Comment:
nit: I would use `removeRole` since this is a per user function and a user only has one role. That said, nothing wrong this either.